### PR TITLE
adding .goreleaser.yml

### LINKED ISSUE #71: new rds exporter release
@maxknee asked in #70 about new release

### PR DESCRIPTION
**Pull request by maxknee**
Adding a `.goreleaser.yml` similarly found `mongodb_exporter`

This should close #71

### CODE REVIEW

**Comment by denisok** [comment]
@maxknee thanks for contribution! looks like you are missing the new line at the end of the file.

Also check this: https://github.com/percona/mongodb_exporter/pull/392  probably would land soon.

**Comment by maxknee** [comment]
> @maxknee thanks for contribution! looks like you are missing the new line at the end of the file.
> 
> Also check this: [percona/mongodb_exporter#392](https://github.com/percona/mongodb_exporter/pull/392) probably would land soon.

do we want to roll this in OR add it after this is merged? Happy to do either

**Comment by denisok** [comment]
> > @maxknee thanks for contribution! looks like you are missing the new line at the end of the file.
> > Also check this: [percona/mongodb_exporter#392](https://github.com/percona/mongodb_exporter/pull/392) probably would land soon.
> 
> do we want to roll this in OR add it after this is merged? Happy to do either

as you would like. just test that it works.

**Comment by maxknee** [comment]
> 

Would prefer to merge, then work on releasing it as a package. 
Need the `aws_iam_role` feature sooner

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  prerelease: auto
+dockers:
+- goos: linux
+  goarch: amd64
+  image_templates:
+  - "percona/rds_exporter:{{.Major}}.{{.Minor}}"
+  - "percona/rds_exporter:{{.Version}}"
+  dockerfile: Dockerfile
+  use: buildx
+
```
Comment (denisok) [inline .goreleaser.yml:43]:
```suggestion

```

**Comment by denisok** [comment]
ok. could you please also add release action then?
https://github.com/percona/mongodb_exporter/blob/main/.github/workflows/release.yml

and please remove docker container as for now, as creating new project in docker hub would take some time and we didn't plan that yet.

or you can try to change it to push to github and lets see if it works: `ghcr.io/` : 
```
  image_templates:
  - "ghcr.io/percona/rds_exporter:{{.Major}}.{{.Minor}}"
  - "ghcr.io/percona/rds_exporter:{{.Version}}"
  ```

**Comment by denisok** [comment]
@maxknee it is and error from Docker file when it tries to build it:
https://github.com/percona/rds_exporter/blob/main/Dockerfile#L9

For new pipeline that is unnecessary and probably doesn't work because there are no sources there (or they are in different location).

You probably need to add new docker file and specify it in the goreleaser:
`dockerfile: Dockerfile.release`

Or rework existing Dockerfile (btw pay attention to the yml, it should be in the container on that path)

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,17 @@
-FROM alpine:latest
```
Comment (denisok) [inline Dockerfile]: looks like you also modified old Docker file. Please leave it untouched.

**Review by denisok — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (denisok) [reply Dockerfile]: oops, some strange cached changes

**Review by denisok — APPROVED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Comment by yonahd** [comment]
@denisok I don't think a release will be able to be created without the tests passing

**Comment by denisok** [comment]
@yonahd those test failed because they couldn't run from the forked repo. We would run test one way or another.

**Comment by yonahd** [comment]
@denisok isn't the goal to create a new release?

**Comment by denisok** [comment]
@yonahd didn't get you. Yes the goal is to create new release. And thus tests need to pass.
https://github.com/percona/rds_exporter/issues/74

**Pull request merged**